subsys/net/ip/net_shell.c

```diff
@@ -1479,10 +1469,7 @@ static int cmd_net_dns_query(const struct shell *shell, size_t argc,
 		}
 	}
 
-	user_data.shell = shell;
-	user_data.user_data = &first;
-
-	ret = dns_get_addr_info(host, qtype, NULL, dns_result_cb, &user_data,
+	ret = dns_get_addr_info(host, qtype, NULL, dns_result_cb, shell,
```

Comment:
i've seen warning `passing argument 5 of ‘dns_get_addr_info’ discards ‘const’ qualifier from pointer` and had to cast shell to `(void *)`

Reply:
> had to cast shell to `(void *)`

Ack, thanks, added it.